lms/djangoapps/course_home_api/outline/v1/tests/test_views.py

```diff
@@ -366,3 +369,39 @@ def test_verified_mode(self):
 
         response = self.client.get(self.url)
         assert response.data['verified_mode'] == {'access_expiration_date': (enrollment.created + MIN_DURATION), 'currency': 'USD', 'currency_symbol': '$', 'price': 149, 'sku': 'ABCD1234', 'upgrade_url': '/dashboard'}
+
+    @override_experiment_waffle_flag(COURSE_HOME_MICROFRONTEND, active=True)
+    def test_hide_learning_sequences(self):
+        """
+        Check that Learning Sequences filters out sequences.
+        """
+        CourseEnrollment.enroll(self.user, self.course.id, CourseMode.VERIFIED)
+
+        # Normal behavior: the sequence exists
+        response = self.client.get(self.url)
+        assert response.status_code == 200
+        blocks = response.data['course_blocks']['blocks']
+        seq_block_id = next(
```

Comment:
TIL! Is this essentially just a more efficient way to do: `[ block_id for .... ][0]`?

Reply:
Yes, though with the caveat that it'll raise a `StopIteration` exception if you hit the end of the generator (in this case if there is no qualifying sequence).